script/prepare_training_environment

```diff
@@ -0,0 +1,188 @@
+    slug: "specialist-publisher",
+    name: "Specialist Publisher",
+    redirect_uri: "http://specialist-publisher.dev.gov.uk/auth/gds/callback",
+    home_uri: "http://specialist-publisher.dev.gov.uk/",
+    description: "Publisher tool for specialist documents",
+    supported_permissions: ["gds_editor"]
+  },
+  {
+    directory_name: "whitehall",
+    slug: "whitehall",
+    name: "Whitehall",
+    redirect_uri: "http://whitehall-admin.dev.gov.uk/auth/gds/callback",
+    home_uri: "http://whitehall-admin.dev.gov.uk/",
+    description: "Whitehall",
+    supported_permissions: ["GDS Editor"]
+  }
+].freeze
+
+puts "Reloading the database schema..."
+`bundle exec rake db:schema:load`
+
+puts "Creating applications and permissions..."
+(TRAINING_PUBLISHING_APPLICATIONS + TRAINING_APIS).each do |application|
+  application_root = "/var/govuk/#{application[:directory_name]}"
+  unless File.directory?(application_root)
+    puts "ERROR #{application[:name]} does not exist in #{application_root}"
+    next
+  end
+
+  doorkeeper_application = Doorkeeper::Application.create!(
+    name: application[:name],
+    redirect_uri: application[:redirect_uri],
+    home_uri: application[:home_uri],
+    description: application[:description]
+  )
+
+  application[:supported_permissions].each do |permission|
+    SupportedPermission.create!(
+      application_id: doorkeeper_application.id,
+      name: permission
+    )
+  end
+end
+
+puts "Creating organisation..."
+organisation = Organisation.create!(
+  slug: "cabinet-office",
+  name: "Cabinet Office",
+  organisation_type: "Ministerial department",
+  abbreviation: "CO",
+  content_id: "96ae61d6-c2a1-48cb-8e67-da9d105ae381"
+)
+
+puts "Creating users..."
+(1..NUMBER_OF_USERS).each do |i|
+  name = "Training User #{i}"
+  email = "training-user-#{i}@digital.cabinet-office.gov.uk"
+  user = User.new(
+    name: name,
+    email: email,
+    password: USER_PASSWORD,
+    password_confirmation: USER_PASSWORD,
+    role: "normal",
+    organisation_id: organisation.id
+  )
+  user.skip_confirmation!
+  user.save!
+end
+
+puts "Creating and assigning API users..."
+TRAINING_APIS.each do |api|
+  name = "API access user for #{api[:name]}"
+  email = "#{api[:slug]}@digital.cabinet-office.gov.uk"
+  user = ApiUser.new(
+    name: name,
+    email: email,
+    password: USER_PASSWORD,
+    password_confirmation: USER_PASSWORD,
+    role: "normal"
+  )
+  user.skip_confirmation!
+  user.api_user = true
+  user.save!
+
+  application = Doorkeeper::Application.where(name: api[:name]).first
+  permissions = SupportedPermission.where(application_id: application.id)
+  permissions.each do |permission|
+    UserApplicationPermission.create!(
+      user_id: user.id,
+      application_id: application.id,
+      supported_permission_id: permission.id
+    )
+  end
+
+  token = "oauth-bearer-token-#{api[:slug]}"
+  ActiveRecord::Base.connection.execute(<<-EOQ)
+    INSERT INTO oauth_access_tokens
```

Comment:
I think this might be the equivalent code from the similar script in the govuk-guix repository https://github.com/alphagov/govuk-guix/blob/c70c9bf4984669cd24aa4c08a921ec6a49f470a6/gds/services/govuk/signon.scm#L264-L271 so you might be able to do this in Ruby.